fgpyo/util/logging.py

```diff
@@ -151,6 +151,22 @@ def record(
         else:
             return False
 
+    def record_alignment(
+        self,
+        rec: AlignedSegment,
+    ) -> bool:
+        """Cerrectly record pysam.AlignedSegments (zero-based coordinates).
+
+        Args:
+            rec: pysam.AlignedSegment object
+
+        Returns:
+            true if a message was logged, false otherwise
+        """
+        position_from_rec = rec.reference_start + 1
+        reference_name_from_rec = rec.reference_name
+        self.record(position=position_from_rec, reference_name=reference_name_from_rec)
```

Comment:
In the issue it implies that `rec.reference_start` can return None - which would cause an error in this function.  Can you please:

1. Add tests for this function that uses `SamBuilder()` to feed it three kinds of records: a) a mapped record, b) an unmapped record with reference_name/start set, c) an unmapped read with no reference_name or reference_start set
2. Fix up this function to ensure those tests pass

Reply:
The function doesn't appear to actually return anything - there's no `return` on this last line.